src/monodroid/jni/inmemory-assemblies.h

```diff
@@ -0,0 +1,91 @@
+#ifndef INC_MONODROID_INMEMORY_ASSEMBLIES_H
+#define INC_MONODROID_INMEMORY_ASSEMBLIES_H
+
+#include "dylib-mono.h"
+#include "jni-wrappers.h"
+#include <string.h>
+
+#define DEFAULT_CAPACITY 8
+
+namespace xamarin { namespace android { namespace internal {
+	class InMemoryAssemblies
+	{
+		private:
+			struct InMemoryAssemblyEntry
+			{
+				int domain_id;
+				int assemblies_count;
+				char **names;
+				char **assemblies_bytes;
+				int *assemblies_bytes_len;
+			};
+
+		public:
+			InMemoryAssemblies ()
+			{
+				capacity = DEFAULT_CAPACITY;
+				entries = new InMemoryAssemblyEntry*[DEFAULT_CAPACITY];
+			}
+
+			bool has_assemblies () const { return length > 0; }
+			MonoAssembly* load_assembly_from_memory (MonoDomain *domain, MonoAssemblyName *name);
+			void add_or_update_from_java (MonoDomain *domain, JNIEnv *env, jstring_array_wrapper &assemblies, jobjectArray assembliesBytes);
+			void clear_for_domain (MonoDomain *domain);
+
+		private:
+			// TODO: replace with some sort of default data structure
+			InMemoryAssemblyEntry **entries;
```

Comment:
Given @grendello's request that this be, in effect, a "desktop-only" file, does this at all mean or imply that we can use the *full* STL in this file?  Meaning this could instead be a `std::vector<InMemoryAssemblyEntry>`?

Reply:
Yep, full STL should be perfectly fine here.

Reply:
I have been going back and forth on this and based on what it pulls down for just that one thing and the extra build changes that would require (cppcompat and friends generate build conflicts otherwise) I would vote for keeping the manual logic unless there is a strong opinion from either of you otherwise.

Reply:
@garuma: this could be resolved by adding `-D__CPP_COMPAT_H` to the Desktop builds, which will prevent the contents of `cppcompat.h` from being evaluated.

Reply:
Yep, the question is do we want to go down the rabbit hole of starting to pull the STL